frontend/pkg/frontend/frontend.go

```diff
@@ -258,12 +261,36 @@ func (f *Frontend) ArmResourceRead(writer http.ResponseWriter, request *http.Req
 
 	// URL path is already lowercased by middleware.
 	resourceID := request.URL.Path
-	cluster, found := f.cache.GetCluster(resourceID)
-	if !found {
-		writer.WriteHeader(http.StatusNotFound)
+	subscriptionID := request.PathValue(PathSegmentSubscriptionID)
+	doc, err := f.dbClient.GetClusterDoc(ctx, resourceID, subscriptionID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
```

Comment:
Our cluster database is still technically a cache and I'm wondering if we should account for cache misses in case the database is somehow incomplete.  Instead of failing here we could ask Cluster Service to search for the cluster with [ClustersListRequest.Search](https://pkg.go.dev/github.com/openshift-online/ocm-sdk-go@v0.1.422/clustersmgmt/v1#ClustersListRequest.Search), and update the cache if we get a hit.  (Probably would want to break this out into a new function since multiple endpoints would need to do this.)

Reply:
Talking about this with @mjlshen  and @bennerv  and @miguelsorianod  on the side. Cosmos is acting as a source of truth here because hitting CS directly without a cluster id could return multiple results back. Using something like the resource name of the cluster may not be unique enough to only get one cluster back. Sounds like CS has new support for using subscription ID and resource group name for queries that might improve that using the search method but even that may not be unique enough if customers spam create the same cluster name in same sub and rg. If they delete the cluster and recreate, it should be removed from CS but might not be right away maybe?

Maybe we leave this for now but i can create a card to make sure we revist this because its for sure something we want to get right but not sure we have all the pieces 100% today? Thoughts?

Reply:
I'm ok with deferring this.  Cluster Service might have their own identification scheme that's name-independent but the cluster still has to be unique to ARM, which identifies resources by subscription ID, resource group name and resource name.  So I think customers can still get name collisions at the ARM level if they're spam-creating clusters like you said.  But yes, we can revisit this after Cluster Service implements their [DDR](https://docs.google.com/document/d/19Yuu-k5IjGW1dac-tSpHDD0KTjPe3gAt_NUvHaTV9Yk/edit).

Reply:
Will get a card created today